### PR TITLE
[Core][Refactor CoreWorker 2/n] split CoreWorkerProcess from CoreWorker

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

Move CoreWorkerProcess from CoreWorker into separate files. This PR depends on https://github.com/ray-project/ray/pull/19675. The follow up PR is #19678

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scv119** [comment]
rlib:test_curiosity is broken on the base commit (#20375)
linux://python/ray/tests:test_output is broken on master
windows is broken on master.

this PR should be good to merge

**Comment by ericl** [comment]
LINT's failing

**Comment by scv119** [comment]
oh no... commented on the wrong PR lol

File: src/ray/core_worker/core_worker_process.cc [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+// limitations under the License.
+
+#include "ray/core_worker/core_worker.h"
+#include "ray/stats/stats.h"
+#include "ray/util/event.h"
+#include "ray/util/util.h"
+
+namespace ray {
+namespace core {
+namespace {
```
Comment (kfstorm) [inline src/ray/core_worker/core_worker_process.cc:22]: What's this empty namespace for?

**Review by kfstorm — APPROVED** [review]
LGTM if they are just movements.

File: src/ray/core_worker/core_worker_process.cc (hunk 1 above)
Comment (scv119) [reply src/ray/core_worker/core_worker_process.cc:22]: https://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces
TL;DR: anonymous namespace is away to avoid ODR issues by defining names local to the current translation unit, similar to what `static` does. This is a [preferred way over static](https://techmunching.com/unnamed-anonymous-namespaces-vs-namespace-static/).
